### PR TITLE
feat: add Option item disableCheckbox

### PR DESCRIPTION
**Pull request by BoyYangzai**
image
- fix https://github.com/ant-design/ant-design/issues/41997

### CODE REVIEW

**Comment by georgeciobanu** [comment]
This is amazing, thank you so much! Should I wait for it to be formally approved or use it as is?

File: src/OptionList/index.tsx [hunk 1]
```diff
@@ -37,6 +37,7 @@
     dropdownPrefixCls,
     loadData,
     expandTrigger,
+    uncheckableItemValues,
```
[the remark this answers is not in the document]
Comment (yoyo837) [reply src/OptionList/index.tsx:40]: How about a callbak function to check it？

File: src/Cascader.tsx [hunk 2]
```diff
@@ -448,6 +450,7 @@ const Cascader = React.forwardRef<CascaderRef, InternalCascaderProps>((props, re
       expandIcon,
       loadingIcon,
       dropdownMenuColumnStyle,
+      uncheckableItemValues,
```
Comment (zombieJ) [inline src/Cascader.tsx:453]: Ref TreeSelect API design. Could you help to modify it as  `disableCheckbox` in options?

**Comment by BoyYangzai** [comment]
> This is amazing, thank you so much! Should I wait for it to be formally approved or use it as is?

This is a new feature, you need to wait until it is officially merged and officially packaged before you can use it 
If you are in a hurry to use it in your project, you can use https://github.com/vercel/ncc to embed the new branch locally if it is not yet released, but I don't recommend doing this😄 @georgeciobanu

**Comment by BoyYangzai** [comment]
@yoyo837 @zombieJ done✅ thanks to cr😄

File: src/OptionList/Column.tsx [hunk 3]
```diff
@@ -59,7 +59,7 @@ export default function Column({
   const optionInfoList = React.useMemo(
     () =>
       options.map(option => {
-        const { disabled } = option;
+        const { disabled, disableCheckbox = false } = option;
```
Comment (yoyo837) [inline src/OptionList/Column.tsx:62]: 这个默认 false 不需要吧?

**Comment by yoyo837** [comment]
如果是不干预选中和勾选状态, 这个属性我个人觉得应该叫 `checkboxClassName` @zombieJ

**Comment by georgeciobanu** [comment]
> > This is amazing, thank you so much! Should I wait for it to be formally approved or use it as is?
> 
> This is a new feature, you need to wait until it is officially merged and officially packaged before you can use it If you are in a hurry to use it in your project, you can use https://github.com/vercel/ncc to embed the new branch locally if it is not yet released, but I don't recommend doing this😄 @georgeciobanu

I really want to try it, it would be super helpful for my work. I tried to use ncc but I don't know how to compile antd using it - any help would be much appreciated.

Alternatively, how long do you think it will take before this feature makes it into an early or stable release of AntD?

For all the amazing AntD developers out there, I am super grateful for your work and absolutely love the component library. Thank you a thousand times!

**Review by zombieJ — APPROVED** [review]

**Pull request merged**